### ISSUE TITLE
Scanning a jitted function throws error in newer JAX versions

### ISSUE THREAD

**Issue by S-Roecken** [google/jax#13419]
### Description

Hi,

First of all thank you for this nice and performant library! I recently set up a new GPU with Jax version==0.3.25, jaxlib=0.3.25+cuda11.cudnn82, though this also happens with jax==0.3.16 and jaxlib=0.3.15+cuda11.cudnn82. When employing jax.lax.scan(jitted_function, state, xs), where jitted_function is a jitted function, the following error is thrown

![Bildschirmfoto 2022-11-28 um 13 20 32](https://user-images.githubusercontent.com/91059576/204278897-af8e1871-6d4c-4af9-aae6-305b1638fe8a.png)

Do you have a fix or a workaround for this? (Not jitting the function would work, but at the cost of not being jitted).  Thank you!

Best
Sebastien

### What jax/jaxlib version are you using?

jax==0.3.25, jaxlib==0.3.15+cuda11.cudnn82

### Which accelerator(s) are you using?

GPU

### Additional system info

Python 3.8.10, Ubuntu 20.04.5 LTS

### NVIDIA GPU info

![Bildschirmfoto 2022-11-28 um 13 38 32](https://user-images.githubusercontent.com/91059576/204279746-9b015746-3112-44e8-9bf1-d86a159b52b9.png)

Nvidia, A100 80GB

**Comment by jakevdp** [comment]
Thanks for the report - could you edit your question to add a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) that demonstrates the issue? It will take the guesswork out as we try to address the problem. Thanks!

**Comment by S-Roecken** [comment]
Thanks for the fast reply - I just tried to set up a minimum working example, however the example does not throw an error and I wasn't able to reproduce the error I get within a larger JaxMD code base.

**Comment by S-Roecken** [comment]
Does passing a normal function to jax.lax.scan(function, state, xs) automatically jit the function? If that's the case, removing jit from the passed function seems like the easiest solution.

Best
Sebastien

**Comment by jakevdp** [comment]
Yes, `scan(f, ...)` will always JIT-compile the input function `f`, unless it is run within a `jax.disable_jit()` context, in which case execution is done via a normal Python for-loop.

**Comment by jakevdp** [comment]
But passing a jit-compiled function to `scan` should be fine, so I would consider it a bug if this leads to an error. If you can supply a code snippet that reproduces this bug, then we can look into it.
